swap/src/lib.rs

```diff
@@ -288,16 +300,20 @@ impl Actor {
                     amount_in,
                     other_receive_amount.as_sat()
                 )
-            })?;
+            })
+            .map_err(|_| InputAmountTooSmall(amount_in, other_receive_amount.as_sat()))?;
```

Comment:
2 things:

1. Unless there is a specific reason not to (like the source error not implementing `StdError`), you should always use `context` over `map_err` otherwise you are dropping the error history.
2. `with_context` is being called right before this, than can go away now.